miner/worker.go

```diff
@@ -661,6 +669,19 @@ func (w *worker) resultLoop() {
 				logs = append(logs, receipt.Logs...)
 			}
 
+			if prev, ok := w.recentMinedBlocks.Get(block.NumberU64()); ok {
+				prevParent, _ := prev.(common.Hash)
+				if prevParent == block.ParentHash() {
+					log.Error("Reject Double Sign!!", "block", block.NumberU64(),
+						"hash", block.Hash(), "root", block.Root(),
+						"ParentHash", block.ParentHash(), "prevParent", prevParent)
+					continue
+				}
+			}
+			// Add will call removeOldest internally to remove the oldest element
+			// if the LRU Cache is full
+			w.recentMinedBlocks.Add(block.NumberU64(), block.ParentHash())
```

Comment:
There is some corner case here:
1. validator propose block based on a parent blockA at height H, it produce BlockB at height H+1;
2. then a possible reorg may happen;
3. validator propose block based on a parent blockA' at height H, it produce BlockB' at height H+1;
4. the cache at H+1 been overwrite;
5. then a possible reorg may happen;
6. validator propose block based on a parent blockA at height H, it produce BlockC at height H+1; this is no detected by the watch dog.

Reply:
It is a bit tricky, but any easy way to solve this?

Reply:
yea, use slice to keep all the history parent blocks, avoid overwritten.